src/main/java/HomePage.java

```diff
@@ -1,35 +1,71 @@
+    private WebElement searchButton;
+
+    @FindBy(xpath = ".//*[@id='header_fullsize']/div/span[2]/div[1]/ul[1]/li[2]/a")
+    private WebElement loginPageButtonLink;
+
+    @FindBy(id = "id_username")
+    private WebElement usernameField;
+
+    @FindBy(id = "id_password")
+    private WebElement passwordField;
+
+    @FindBy(id = "login-btn")
+    private WebElement loginButton;
 
-    public ResultPage searchCityName(String cityName) {
-        driver.findElement(By.id(Config.getSetting("search_field"))).sendKeys(cityName);
-        driver.findElement(By.xpath(Config.getSetting("search_button"))).click();
+    @FindBy(xpath = ".//*[@id='header_fullsize']/div[1]/span[2]/div/div/div[1]/span")
+    private WebElement logoutSpan;
 
-        return new ResultPage(driver);
+    @FindBy(xpath = ".//*[@id='header_fullsize']/div[1]/span[2]/div/div/ul/li[5]/a")
+    private WebElement logoutButton;
+
+    @FindBy(xpath = ".//*[@id='header_fullsize']/div[1]/span[2]/div/span")
+    private WebElement successfulLogin;
+
+    // TODO: move crypto to correct place
+    private String decrypt(String encryptedString) {
+        byte[] byteArray = Base64.decodeBase64(encryptedString.getBytes());
+        return new String(byteArray);
+    }
+
+    // TODO: need to resolve the issue with not used return values
+    public SearchResultPage searchCityByName() {
+        searchField.sendKeys(Config.getSetting("testCity"));
+        searchButton.click();
+        return new SearchResultPage();
     }
 
-    public LoginPage logInToParkMe(String login, String password) {
-        driver.findElement(By.xpath(".//*[@id='header_fullsize']/div/span[2]/div[1]/ul[1]/li[2]/a")).click();
-        driver.findElement(By.id("id_username")).sendKeys(login);
-        driver.findElement(By.id("id_password")).sendKeys(password);
-        driver.findElement(By.id("login-btn")).click();
+    public boolean checkifLoginCorrect() {
+        return successfulLogin.getText().contains(decrypt(Config.getSetting("correctUsernameBase64")));
+    }
 
-        return new LoginPage(driver);
+    // TODO: need to move login logout method to Page class as it is static
+    public void login() {
+        loginPageButtonLink.click();
+        usernameField.sendKeys(decrypt(Config.getSetting("correctUsernameBase64")));
+        passwordField.sendKeys(decrypt(Config.getSetting("correctPasswordBase64")));
+        loginButton.click();
     }
 
-    public LoginPage logOutFromParkMe() {
-        driver.findElement(By.xpath(".//*[@id='header_fullsize']/div[1]/span[2]/div/div/div[1]/span")).click();
-        driver.findElement(By.xpath(".//*[@id='header_fullsize']/div[1]/span[2]/div/div/ul/li[5]/a")).click();
+    public void login_negative() {
+        loginPageButtonLink.click();
+        usernameField.sendKeys(Config.getSetting("incorrectUsername"));
+        passwordField.sendKeys(Config.getSetting("incorrectPassword"));
+        loginButton.click();
+    }
```

Comment:
should be one login method which accepts two parameters: usrname and pwd, so tester can define what exactly he wants to test

Reply:
fixed on new commits